runhouse/resources/functions/aws_lambda.py

```diff
@@ -0,0 +1,889 @@
+            system=self.RESOURCE_TYPE,
+            env=env,
+            fn_pointers=fn_pointers,
+            access=access or self.DEFAULT_ACCESS,
+            **kwargs,
+        )
+
+        self.local_path_to_code = paths_to_code
+        self.handler_function_name = handler_function_name
+        self.args_names = args_names
+        self.runtime = runtime
+        self.timeout = timeout
+        self.memory_size = memory_size
+        self.tmp_size = tmp_size
+        self.aws_lambda_config = (
+            None  # Lambda config and role arn from aws will be saved here
+        )
+        self.retention_time = retention_time
+        self.function_arn = function_arn
+
+        # will be used for reloading shared functions from other regions.
+        if function_arn:
+            region = function_arn.split(":")[3]
+            self.lambda_client = boto3.client("lambda", region_name=region)
+            self.logs_client = boto3.client("logs", region_name=region)
+        else:
+
+            if Path(CRED_PATH).is_file():
+                self.lambda_client = boto3.client("lambda")
+                self.logs_client = boto3.client("logs")
+
+            else:
+                self.lambda_client = boto3.client(
+                    "lambda", region_name=self.DEFAULT_REGION
+                )
+                self.logs_client = boto3.client("logs", region_name=self.DEFAULT_REGION)
+
+    # --------------------------------------
+    # Constructor helper methods
+    # --------------------------------------
+
+    @classmethod
+    def from_config(cls, config: dict, dryrun: bool = False):
+        """Create an AWS Lambda object from a config dictionary."""
+
+        if "resource_subtype" in config.keys():
+            config.pop("resource_subtype", None)
+        if "system" in config.keys():
+            config.pop("system", None)
+        keys = config.keys()
+        if "fn_pointers" not in keys:
+            config["fn_pointers"] = None
+        if "timeout" not in keys:
+            config["timeout"] = AWSLambdaFunction.DEFAULT_TIMEOUT
+        if "memory_size" not in keys:
+            config["memory_size"] = AWSLambdaFunction.DEFAULT_MEMORY_SIZE
+        if "env" not in keys:
+            config["env"] = Env(
+                reqs=[],
+                env_vars={"HOME": AWSLambdaFunction.HOME_DIR},
+                name=Env.DEFAULT_NAME,
+            )
+        else:
+            config["env"] = Env.from_config(config["env"])
+
+        return AWSLambdaFunction(**config, dryrun=dryrun).to()
+
+    @classmethod
+    def from_name(cls, name, dryrun=False, alt_options=None):
+        config = rns_client.load_config(name=name)
+        if not config:
+            raise ValueError(f"Could not find a Lambda called {name}.")
+        return cls.from_config(config)
+
+    # --------------------------------------
+    # Private helping methods
+    # --------------------------------------
+
+    @classmethod
+    def _reqs_to_list(cls, env):
```

Comment:
Maybe add a note that this should no longer be needed once we send the `Env` object to the cluster?

Reply:
removed this function, since we are installing requirements.txt during runtime.